signature/openpgp/openpgp.go

```diff
@@ -0,0 +1,86 @@
+package openpgp
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"time"
+
+	"github.com/containers/image/types"
+
+	"golang.org/x/crypto/openpgp"
+)
+
+type context struct {
+	keyring openpgp.EntityList
+}
+
+type OpenPGPMechanism struct {
+	ctx *context
+}
+
+func NewOpenPGPSigningMechanism() (types.SigningMechanism, error) {
+	return OpenPGPMechanism{ctx: &context{
+		keyring: openpgp.EntityList{},
+	}}, nil
+}
+
+func (m OpenPGPMechanism) ImportKeysFromBytes(blob []byte) ([]string, error) {
+	keyring, err := openpgp.ReadArmoredKeyRing(bytes.NewReader(blob))
```

Comment:
(These kinds of things have so far not been really defined because “whatever GPG does” has worked for us; we will need to gradually nail them down now.)